### ISSUE TITLE
Explicitly indexed variables don't bind correctly on Android

### ISSUE THREAD

**Issue by simolus3** [tekartik/sqflite#272]
The following code snippet:
```dart
  final db = await openDatabase(':memory:');
  print(await db.rawQuery('SELECT ?1 + ?2', [3, 4]));
  await db.close();
```

Prints `[{31 + 42: 73}]` on Android, while it should output `[3 + 4: 7]`.

**Comment by alextekartik** [comment]
Thanks for catching this. Indeed due to some another bug on Android (https://github.com/tekartik/sqflite/issues/64) I do replace the parameters of type int manually. I have to admit that I was not aware of this type of param. I look into fixing that.

Thanks for the simple repro case

**Comment by simolus3** [comment]
Wow, thanks for the quick reply! I just tried to fix this locally with
```java
    public SqlCommand sanitizeForQuery() {
        if (rawArguments.size() == 0) {
            return this;
        }
        StringBuilder sanitizeSqlSb = new StringBuilder();
        List<Object> sanitizeArguments = new ArrayList<>();
        int count = 0;
        int sqlLength = sql.length();
        for (int i = 0; i < sqlLength; i++) {
            char ch = sql.charAt(i);
            if (ch == '?') {
                // Read explicit index, if one exists
                StringBuilder explicitIndexBuilder = null;
                // consume digits that follow this question mark
                while (i != sqlLength - 1) {
                    char next = sql.charAt(i + 1);
                    if ('0' <= next && next <= '9') {
                        if (explicitIndexBuilder == null) {
                            explicitIndexBuilder = new StringBuilder();
                        }
                        explicitIndexBuilder.append(next);
                        i++;
                    } else {
                       break;
                    }
                }

                int indexOfThisVariable; // starts at one in sqlite
                if (explicitIndexBuilder != null) {
                    indexOfThisVariable = Integer.parseInt(explicitIndexBuilder.toString());
                    // after an explicitly indexed variable, the next variable will still have this
                    // index plus one.
                    count = indexOfThisVariable;
                } else {
                    indexOfThisVariable = ++count;
                }

                int argumentIndex = indexOfThisVariable - 1;
                // no match, return the same
                if (argumentIndex >= rawArguments.size()) {
                    return this;
                }
                Object argument = rawArguments.get(argumentIndex);
                if (argument instanceof Integer || argument instanceof Long) {
                    sanitizeSqlSb.append(argument.toString());
                    continue;
                } else {
                    // Let the other args as is
                    sanitizeArguments.add(argument);
                }
            }
            // Simply append the existing
            sanitizeSqlSb.append(ch);
        }
        // no match (there might be an extra ? somwhere), return the same
        if (count != rawArguments.size()) {
            return this;
        }
        return new SqlCommand(sanitizeSqlSb.toString(), sanitizeArguments);
    }
```

I'm still fighting gradle so I couldn't test this yet, but maybe it helps as a starting point.

**Comment by simolus3** [comment]
I guess your fix works as well. Again, thank you for this great project and for handling this so quickly!

**Comment by alextekartik** [comment]
I just saw your notes...I took a simple approach and simply cancel the conversion when we have indexed parameters, sounds like a safe bet for now.

To note that I do get `{'?1 + ?2': 7}` as a result. I tested on both iOS and Android.

Fixed in 1.1.6+5 just published. While I have not tried your moor project yet (I tend to write raw sql), I'm  following what you do, especially on ffi! Thanks for the report!

**Comment by simolus3** [comment]
Thank you for fixing this so fast!
> To note that I do get `{'?1 + ?2': 7}` as a result. I tested on both iOS and Android.

That's fair. I initially assumed that integers were still inlined for explicit variables, but not doing that works fine. 
> I'm following what you do, especially on ffi!

Thanks again :) We're still tackling some problems (basically the ones you outlined in https://github.com/tekartik/sqflite/issues/200#issuecomment-532662550), but I've made some good progress on binary distribution last week. After we have something that works reliably, I can share how we approached these problems if you're looking into ffi as well.
